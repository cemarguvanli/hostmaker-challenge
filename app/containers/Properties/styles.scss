.propertiesPage {
  padding-bottom: 32px;
}


.introduce {
  display: flex;
  flex-direction: column;
  text-align: center;
  justify-content: center;

  min-height: 200px;
  color: #fff;
  padding: 0 16px;

  background-repeat: no-repeat;
  background-size: cover;
  background-position: top;
  background-image: url(https://hostmaker-website.imgix.net/assets/images/services/hero/investors.jpg);

  h1 {
    font-size: 32px;
    line-height: 36px;
    font-weight: 500;
    padding: 0px 8px 24px;
    margin: 0;

    @media (max-width: 768px) {
      font-size: 24px;
    }
  }
  p {
    margin: 0 auto;
    max-width: 640px;
  }
}