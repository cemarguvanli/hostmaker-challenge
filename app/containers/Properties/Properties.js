import React from 'react';

import layoutStyles from 'common/styles/layout.scss';

import PropertyTable from './views/PropertyTable';

import data from '../../data.json';
import styles from './styles.scss';

const PropertyList = () => (
  <div className={styles.propertiesPage}>
    <div className={styles.introduce}>
      <h1>Property details list</h1>
      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>
    </div>
    <div className={layoutStyles.container}>
      <PropertyTable
        properties={data}
      />
    </div>
  </div>
);

export default PropertyList;
