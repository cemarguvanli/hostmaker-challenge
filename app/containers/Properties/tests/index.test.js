import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Properties from '../Properties';

describe('<Properties />', () => {
  it('should render the properties', () => {
    const wrapper = shallow(<Properties />);
    expect(wrapper).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Properties />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
