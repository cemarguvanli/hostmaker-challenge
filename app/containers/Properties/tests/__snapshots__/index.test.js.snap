// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Properties /> renders correctly 1`] = `
<div>
  <div>
    <h1>
      Property details list
    </h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
  </div>
  <div>
    <div>
      <table>
        <thead>
          <tr>
            <th>
              Owner
            </th>
            <th>
              Address
            </th>
            <th>
              IncomeGenerated
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Carlos
            </td>
            <td>
              <p>
                Flat 5
              </p>
              <p>
                7 Westbourne Terrace
              </p>
              <p>
                W2 3UL
              </p>
              <p>
                London
              </p>
              <p>
                U.K.
              </p>
            </td>
            <td>
              £
              2000.34
            </td>
          </tr>
          <tr>
            <td>
              Ankur
            </td>
            <td>
              <p>
                4
              </p>
              <p>
                Tower Mansions
              </p>
              <p>
                Off Station road
              </p>
              <p>
                86-87 Grange Road
              </p>
              <p>
                SE1 3BW
              </p>
              <p>
                London
              </p>
              <p>
                U.K.
              </p>
            </td>
            <td>
              £
              10000
            </td>
          </tr>
          <tr>
            <td>
              Elaine
            </td>
            <td>
              <p>
                4
              </p>
              <p>
                332b
              </p>
              <p>
                Goswell Road
              </p>
              <p>
                EC1V 7LQ
              </p>
              <p>
                London
              </p>
              <p>
                U.K.
              </p>
            </td>
            <td>
              £
              1200
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
